File: Project_Integration_Overview (1)/labl_iq_analysis/dashboard/services/shipping.ts
// Shipping analysis service
interface ShipmentData {
  id: string;
  origin: {
    zipCode: string;
    city: string;
    state: string;
  };
  destination: {
    zipCode: string;
    city: string;
    state: string;
  };
  weight: number;
  dimensions: {
    length: number;
    width: number;
    height: number;
  };
  value: number;
  service: string;
  carrier: string;
  actualCost: number;
  shipDate: string;
}

interface AnalysisResult {
  id: string;
  uploadId: string;
  totalShipments: number;
  totalCost: number;
  potentialSavings: number;
  avgSavingsPerShipment: number;
  topSavingsOpportunities: {
    carrier: string;
    service: string;
    currentCost: number;
    recommendedCost: number;
    savings: number;
  }[];
  carrierBreakdown: {
    carrier: string;
    shipments: number;
    cost: number;
    savings: number;
  }[];
  zoneAnalysis: {
    zone: string;
    shipments: number;
    avgCost: number;
    savings: number;
  }[];
  createdAt: string;
  status: 'processing' | 'completed' | 'failed';
}

interface DashboardMetrics {
  totalShipments: number;
  totalSavings: number;
  avgSavingsPerShipment: number;
  topCarrier: string;
  recentAnalyses: AnalysisResult[];
  monthlyTrends: {
    month: string;
    shipments: number;
    cost: number;
    savings: number;
  }[];
}

interface RateQuote {
  carrier: string;
  service: string;
  cost: number;
  deliveryDays: number;
  deliveryDate: string;
}

class ShippingService {
  async uploadShipmentData(file: File, options?: { 
    carriersToAnalyze?: string[];
    includeInternational?: boolean;
  }): Promise<{ uploadId: string; message: string }> {
    const response = await apiClient.uploadFile<{ uploadId: string; message: string }>(
      '/shipping/upload',
      file,
      options ? {
        carriersToAnalyze: JSON.stringify(options.carriersToAnalyze || []),
        includeInternational: String(options.includeInternational || false)
      } : undefined
    );

    if (response.success) {
      return response.data;
    }

    throw new Error(response.message || 'Upload failed');
  }

  async getAnalysisStatus(uploadId: string): Promise<AnalysisResult> {
    const response = await apiClient.get<AnalysisResult>(`/shipping/analysis/${uploadId}`);
    
    if (response.success) {
      return response.data;
    }
    
    throw new Error(response.message || 'Failed to get analysis status');
  }

  async getAnalysisResults(analysisId: string): Promise<AnalysisResult> {
    const response = await apiClient.get<AnalysisResult>(`/shipping/results/${analysisId}`);
    
    if (response.success) {
      return response.data;
    }
    
    throw new Error(response.message || 'Failed to get analysis results');
  }

  async getDashboardMetrics(timeframe: '7d' | '30d' | '90d' = '30d'): Promise<DashboardMetrics> {
    const response = await apiClient.get<DashboardMetrics>(`/shipping/dashboard?timeframe=${timeframe}`);
    
    if (response.success) {
      return response.data;
    }
    
    throw new Error(response.message || 'Failed to get dashboard metrics');
  }

  async getRateQuotes(shipment: {
    origin: string;
    destination: string;
    weight: number;
    dimensions: { length: number; width: number; height: number };
    value?: number;
  }): Promise<RateQuote[]> {
    const response = await apiClient.post<RateQuote[]>('/shipping/quotes', shipment);
    
    if (response.success) {
      return response.data;
    }
    
    throw new Error(response.message || 'Failed to get rate quotes');
  }

  async exportAnalysis(analysisId: string, format: 'csv' | 'xlsx' | 'pdf' = 'xlsx'): Promise<Blob> {
    const url = `${apiClient['baseURL']}/shipping/export/${analysisId}?format=${format}`;
    const response = await fetch(url, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem('auth_token')}`,
      },
    });

    if (!response.ok) {
      throw new Error('Export failed');
    }

    return response.blob();
  }

  async deleteAnalysis(analysisId: string): Promise<void> {
    const response = await apiClient.delete(`/shipping/analysis/${analysisId}`);
    
    if (!response.success) {
      throw new Error(response.message || 'Failed to delete analysis');
    }
  }
}

export const shippingService = new ShippingService();
export type { 
  ShipmentData, 
  AnalysisResult, 
  DashboardMetrics, 
  RateQuote 
};

File: ARCHIVE_OLD_VERSIONS/Project_Integration_Overview (1)/labl_iq_analysis/dashboard/backend/app/schemas/shipping.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime
from enum import Enum

class AnalysisStatus(str, Enum):
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class UploadResponse(BaseModel):
    uploadId: str
    message: str

class CarrierBreakdown(BaseModel):
    carrier: str
    shipments: int
    cost: float
    savings: float

class ZoneAnalysis(BaseModel):
    zone: str
    shipments: int
    avgCost: float
    savings: float

class SavingsOpportunity(BaseModel):
    carrier: str
    service: str
    currentCost: float
    recommendedCost: float
    savings: float

class AnalysisResult(BaseModel):
    id: str
    uploadId: str
    totalShipments: int
    totalCost: float
    potentialSavings: float
    avgSavingsPerShipment: float
    topSavingsOpportunities: List[SavingsOpportunity]
    carrierBreakdown: List[CarrierBreakdown]
    zoneAnalysis: List[ZoneAnalysis]
    createdAt: datetime
    status: AnalysisStatus

class MonthlyTrend(BaseModel):
    month: str
    shipments: int
    cost: float
    savings: float

class DashboardMetrics(BaseModel):
    totalShipments: int
    totalSavings: float
    avgSavingsPerShipment: float
    topCarrier: str
    recentAnalyses: List[AnalysisResult]
    monthlyTrends: List[MonthlyTrend]

class RateQuoteRequest(BaseModel):
    origin: str
    destination: str
    weight: float
    dimensions: Dict[str, float]  # length, width, height
    value: Optional[float] = None

class RateQuote(BaseModel):
    carrier: str
    service: str
    cost: float
    deliveryDays: int
    deliveryDate: str